src/org/jitsi/impl/neomedia/rtp/translator/SimulcastController.java

```diff
@@ -147,6 +147,20 @@ public long getTargetSSRC()
      */
     public boolean accept(byte[] buf, int off, int len)
     {
+        int targetIndex = bitstreamController.getTargetIndex()
+            , currentIndex = bitstreamController.getCurrentIndex();
+
+        if (targetIndex < 0 && currentIndex > -1)
+        {
+            synchronized (this)
```

Comment:
We want to keep some synchronization when we switch from one bitstream controller to another so that the target and the optimal index are not messed up. What changed here is that we only write the this.bitstreamController field from the filter thread.